backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/service/impl/InterimInvoiceFlatrateTermBL.java

```diff
@@ -0,0 +1,192 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contacts.invoice.interim.service.impl;
+
+import ch.qos.logback.classic.Level;
+import com.google.common.collect.ImmutableSet;
+import de.metas.bpartner.BPartnerId;
+import de.metas.contacts.invoice.interim.InterimInvoiceFlatrateTerm;
+import de.metas.contacts.invoice.interim.command.InterimInvoiceFlatrateTermCreateCommand;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceFlatrateTermBL;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceFlatrateTermDAO;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceFlatrateTermLineDAO;
+import de.metas.contracts.ConditionsId;
+import de.metas.contracts.IFlatrateDAO;
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.inout.IInOutDAO;
+import de.metas.inout.InOutAndLineId;
+import de.metas.inout.InOutLineId;
+import de.metas.invoicecandidate.InvoiceCandidateId;
+import de.metas.invoicecandidate.InvoiceCandidateIds;
+import de.metas.invoicecandidate.api.IInvoiceCandDAO;
+import de.metas.invoicecandidate.model.I_C_Invoice_Candidate;
+import de.metas.logging.LogManager;
+import de.metas.order.OrderLineId;
+import de.metas.product.ProductId;
+import de.metas.quantity.Quantity;
+import de.metas.quantity.Quantitys;
+import de.metas.uom.UomId;
+import de.metas.util.Loggables;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.I_M_InOutLine;
+import org.slf4j.Logger;
+
+import java.math.BigDecimal;
+
+public class InterimInvoiceFlatrateTermBL implements IInterimInvoiceFlatrateTermBL
+{
+	private final IInOutDAO inOutDAO = Services.get(IInOutDAO.class);
+	private final IInterimInvoiceFlatrateTermDAO interimInvoiceFlatrateTermDAO = Services.get(IInterimInvoiceFlatrateTermDAO.class);
+	private final IInterimInvoiceFlatrateTermLineDAO interimInvoiceFlatrateTermLineDAO = Services.get(IInterimInvoiceFlatrateTermLineDAO.class);
+	private final IFlatrateDAO flatrateDAO = Services.get(IFlatrateDAO.class);
+	private final IInvoiceCandDAO invoiceCandDAO = Services.get(IInvoiceCandDAO.class);
+
+	private static final Logger logger = LogManager.getLogger(InterimInvoiceFlatrateTermBL.class);
+
+	@Override
+	public void updateInterimInvoiceFlatrateTermForInOutLine(@NonNull final I_M_InOutLine inOutLine)
+	{
+		InterimInvoiceFlatrateTerm interimInvoiceFlatrateTerm = interimInvoiceFlatrateTermDAO.getInterimInvoiceOverviewForInOutLine(inOutLine);
+		if (interimInvoiceFlatrateTerm == null)
```

Comment:
please also check if the term is actually completed

Reply:
DMed, agreed to put a comment stating this.